## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 470973`
Commit: `604bda8544b4a5e2cc003382c451e012e1ed6d77`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.xclim</li>
    <li>python312Packages.xclim.dist</li>
    <li>python313Packages.xclim</li>
    <li>python313Packages.xclim.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.xclim</li>
    <li>python312Packages.xclim.dist</li>
    <li>python313Packages.xclim</li>
    <li>python313Packages.xclim.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.xclim</li>
    <li>python312Packages.xclim.dist</li>
    <li>python313Packages.xclim</li>
    <li>python313Packages.xclim.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.xclim</li>
    <li>python312Packages.xclim.dist</li>
    <li>python313Packages.xclim</li>
    <li>python313Packages.xclim.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.xclim</summary>
<pre>Rewriting #!/nix/store/fdibxyh7xcmqrc172y78awzhxs292gq1-python3-3.12.12/bin/python3.12 to #!/nix/store/fdibxyh7xcmqrc172y78awzhxs292gq1-python3-3.12.12
wrapping `/nix/store/56nz1ccziq5pc8ccvacmr7bmcl0pw0ik-python3.12-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installCheckPhase&quot; }
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/56nz1ccziq5pc8ccvacmr7bmcl0pw0ik-python3.12-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.12/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
<details>
<summary>python313Packages.xclim</summary>
<pre>Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installCheckPhase&quot; }
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/3g61v1ardmjgqxnns5dfhp32r6dp2s0v-python3.13-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.13/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
    ...&lt;9 lines&gt;...
    )
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.xclim</summary>
<pre>patching script interpreter paths in /nix/store/pn4qj6czdszcb6r06xcdgp1c5xjxiniy-python3.12-xclim-0.59.1-dist
Rewriting #!/nix/store/yyns9w3mfdxn5jwqa3lbp461wq9nl5p7-python3-3.12.12/bin/python3.12 to #!/nix/store/yyns9w3mfdxn5jwqa3lbp461wq9nl5p7-python3-3.12.12
wrapping `/nix/store/byd5wdby0952v8lq08jvb99siicdapvb-python3.12-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/byd5wdby0952v8lq08jvb99siicdapvb-python3.12-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.12/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
<details>
<summary>python313Packages.xclim</summary>
<pre>wrapping `/nix/store/kk2impr01m1b38cnvsnarflfb7qbci60-python3.13-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/kk2impr01m1b38cnvsnarflfb7qbci60-python3.13-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.13/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
    ...&lt;9 lines&gt;...
    )
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.xclim</summary>
<pre>Rewriting #!/nix/store/978h7w9sbz3f6qi6ra4pp6prqflgqyqq-python3-3.12.12/bin/python3.12 to #!/nix/store/978h7w9sbz3f6qi6ra4pp6prqflgqyqq-python3-3.12.12
wrapping `/nix/store/7xq34wjf5b9ard1x5dj3psd9s44chrcv-python3.12-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
fixupPhase completed in 1 minutes 49 seconds
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/7xq34wjf5b9ard1x5dj3psd9s44chrcv-python3.12-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.12/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
<details>
<summary>python313Packages.xclim</summary>
<pre>wrapping `/nix/store/yy61da9s9j10xxnhsg3br7zgs1kbjmgl-python3.13-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/yy61da9s9j10xxnhsg3br7zgs1kbjmgl-python3.13-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.13/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
    ...&lt;9 lines&gt;...
    )
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.xclim</summary>
<pre>patching script interpreter paths in /nix/store/yf3mbp3p7rhdsix23y9mwnr59b9zyx7r-python3.12-xclim-0.59.1-dist
Rewriting #!/nix/store/nrk1773m026hlgkk7vlpvc2s6zlwf1rh-python3-3.12.12/bin/python3.12 to #!/nix/store/nrk1773m026hlgkk7vlpvc2s6zlwf1rh-python3-3.12.12
wrapping `/nix/store/hggz7cc5vycylwklrf1qdq9k5gzbanrx-python3.12-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/hggz7cc5vycylwklrf1qdq9k5gzbanrx-python3.12-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.12/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.12/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
<details>
<summary>python313Packages.xclim</summary>
<pre>wrapping `/nix/store/rwswzcxy6f5qy4652a9qp0j4k8fkpgkz-python3.13-xclim-0.59.1/bin/xclim&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Did not find version 0.59.1 in the output of the command /nix/store/rwswzcxy6f5qy4652a9qp0j4k8fkpgkz-python3.13-xclim-0.59.1/bin/xclim --version
{{storeDir}}/lib/python3.13/site-packages/xclim/core/units.py:19: UserWarning: Import(s) unavailable to set up matplotlib support...skipping this portion of the setup.
  import cf_xarray.units
Traceback (most recent call last):
  File &quot;{{storeDir}}/bin/.xclim-wrapped&quot;, line 6, in &lt;module&gt;
    from xclim.cli import cli
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/cli.py&quot;, line 20, in &lt;module&gt;
    from xclim.testing.utils import (
    ...&lt;9 lines&gt;...
    )
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/__init__.py&quot;, line 5, in &lt;module&gt;
    from . import helpers
  File &quot;{{storeDir}}/lib/python3.13/site-packages/xclim/testing/helpers.py&quot;, line 11, in &lt;module&gt;
    import pooch
ModuleNotFoundError: No module named &#x27;pooch&#x27;</pre>
</details>
</details>

--- a/pkgs/development/python-modules/xclim/default.nix
+++ b/pkgs/development/python-modules/xclim/default.nix
@@ -1,0 +1,99 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  flit,
+  setuptools,
+
+  # dependencies
+  boltons,
+  bottleneck,
+  cf-xarray,
+  cftime,
+  click,
+  dask,
+  filelock,
+  jsonpickle,
+  numba,
+  packaging,
+  pandas,
+  pint,
+  platformdirs,
+  pooch,
+  pyarrow,
+  pyyaml,
+  scikit-learn,
+  scipy,
+  statsmodels,
+  xarray,
+  yamale,
+
+  # test
+  versionCheckHook,
+}:
+buildPythonPackage rec {
+  pname = "xclim";
+  version = "0.59.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Ouranosinc";
+    repo = "xclim";
+    tag = "v${version}";
+    hash = "sha256-n9HJoIHLyLWxrgCuDZDQ9dcW7frgEA/LoYqnTEBLqD8=";
+  };
+
+  build-system = [
+    flit
+    setuptools
+  ];
+
+  dependencies = [
+    boltons
+    bottleneck
+    cf-xarray
+    cftime
+    click
+    dask
+    filelock
+    jsonpickle
+    numba
+    packaging
+    pandas
+    pint
+    platformdirs
+    pooch
+    pyarrow
+    pyyaml
+    scikit-learn
+    scipy
+    statsmodels
+    xarray
+    yamale
+  ];
+
+  # No python test hooks has been added as all tests seems to be relying on network data
+  # https://github.com/Ouranosinc/xclim/blob/e8ce9bf37083832517afb3375acc853191782d8f/tests/conftest.py#L314
+  nativeCheckInputs = [
+    versionCheckHook
+  ];
+
+  versionCheckProgramArg = "--version";
+
+  pythonImportsCheck = [
+    "xclim"
+    "xclim.ensembles"
+    "xclim.indices"
+  ];
+
+  meta = {
+    description = "Operational Python library supporting climate services, based on xarray";
+    homepage = "https://github.com/Ouranosinc/xclim";
+    changelog = "https://github.com/Ouranosinc/xclim/releases/tag/${src.tag}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ daspk04 ];
+    mainProgram = "xclim";
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -20653,6 +20653,8 @@ self: super: with self; {
 
   xcffib = callPackage ../development/python-modules/xcffib { };
 
+  xclim = callPackage ../development/python-modules/xclim { };
+
   xdg = callPackage ../development/python-modules/xdg { };
 
   xdg-base-dirs = callPackage ../development/python-modules/xdg-base-dirs { };

